documentation/src/main/asciidoc/user_guide/clustering.adoc

```diff
@@ -1,211 +1,308 @@
+backup will see the new value, but reads from the primary will see the old one.
 
-Each hash segment is mapped to a list of nodes called *owners*.
-The order matters, because the first owner, also known as the *primary owner*, has a
-special role in many cache operations (e.g. locking).
-The other owners are called *backup owners*.
-There is no hard rule on how the segments are mapped to owners, although the hashing
-algorithms generally try to balance the number of segments allocated to each node and at
-the same time minimize the number of segments that have to move after a node joins or
-leaves the cluster.
+[[key_ownership]]
+==== Key ownership
+Distributed caches split entries into a fixed number of segments, and assign each segment
+to a list of owner nodes.
+Replicated caches do the same, except every node is an owner.
 
-The hashing algorithm in Infinispan is customizable, and in fact there are five
-implementations that ship with Infinispan by default:
+The first node in the owners list is called the *primary owner*, and the others are
+called *backup owners*.
+The segment ownership table is broadcast to every node when the cache topology changes
+(i.e. a node joins or leaves the cluster).
+This way, a node can compute the location of a key itself, without resorting to multicast
+requests or maintaining per-key metadata.
 
-org.infinispan.distribution.ch.DefaultConsistentHashFactory::
-  The default hashing algorithm.
+The number of segments is configurable (`numSegments`), but it cannot be changed without
+restarting the cluster.
+The mapping of keys to segments is also fixed -- a key must map to the same segment,
+regardless of how the topology of the cluster changes.
+The key-to-segment mapping can be customized by configuring a
+link:https://docs.jboss.org/infinispan/{infinispanversion}/apidocs/org/infinispan/distribution/ch/KeyPartitioner.html[KeyPartitioner]
+or by using the <<grouping,Grouping API>>.
+
+There is no hard rule on how segments must be mapped to owners, but the goal is to
+balance the number of segments allocated to each node and at the same time minimize the
+number of segments that have to move after a node joins or leaves the cluster.
+The segment mapping is customizable, and in fact there are five implementations that
+ship with Infinispan:
+
+SyncConsistentHashFactory::
+  An algorithm based on
+  link:http://en.wikipedia.org/wiki/Consistent_hashing[consistent hashing].
+  It always assigns a key to the same node in every cache as long as the cluster is
+  symmetric (i.e. all caches run on all nodes).
+  It does have some weaknesses: the load distribution is a bit uneven, and it also moves
+  more segments than strictly necessary on a join or leave.
+TopologyAwareSyncConsistentHashFactory::
+  Selected automatically when <<ServerHinting, Server Hinting>> is enabled.
+  Similar to _SyncConsistentHashFactory_, but adapted for
+  <<ServerHinting, Server Hinting>>.
+DefaultConsistentHashFactory::
+  The old default.
```

Comment:
The doesn't sound good... Maybe something like - "using prior to Infinispan X" ?